migration/mule/jms-message-rollback-and-redelivery/README.md

```diff
@@ -0,0 +1,67 @@
+# JMS message rollback and redelivery
+
+This example shows how to implement jms rollback and redelivery within the WSO2 Integration Studio. While when we passing the jms messages inside the WSO2 EI, following failures could be happening.
+* Failures during message mediation process.
+* Message sending failure,due to endpoint is unavailability.
+
+The JMS message passing flow in this example is configured with the ActiveMQ MOM. The requirement is if any of the above failures happen, the message should not process further and send to end point, rather it should roll back. After a specific number of unsuccessful failures happen the message should send to ActiveMQ topic.
+## Assumptions
+
+This document describes the details of the example within the context of WSO2 Integration Studio, editor to develop WSO2 EI artifacts. This document assumes that you are familiar with WSO2 EI and the [Integration Studio interface](https://ei.docs.wso2.com/en/latest/micro-integrator/develop/WSO2-Integration-Studio/). To increase your familiarity with Integration Studio, consider completing one or more [WSO2 EI Tutorials](https://ei.docs.wso2.com/en/latest/micro-integrator/use-cases/integration-use-cases/).
+
+## Example Use Case
+
+In this example, there is a JMS message consuming inbound endpoint implemented inside WSO2 EI. It listens to the ActiveMQ Queue. After consuming the JMS message by the Inbound endpoint, which rollbacks and retries to deliver the same message to the topic reside in the ActiveMQ. This rollback process is handled by `OnErrorPropagate` sequence.  
```

Comment:
This statement is wrong. `there is a JMS message consuming inbound endpoint implemented inside WSO2 EI`. I do not get what is meant by that. 

Shall we say, 

In this example there is a JMS message in transaction inside WSO2 EI mediation flow which throws an exception when processed (`makefault` mediator is used to generate the error). This message is then handled by the `fault sequence`, which rollbacks and retries to deliver the same message. After a specific number of unsuccessful attempts to commit (4 in this case), it sends the message successfully.